File: interview-main/src/javascript/fizz_buzz/fizz_buzz.test.js
const fizzBuzz = require("./fizz_buzz");

describe("existance of functions:", () => {
    test("defined", () => {
        expect(fizzBuzz).toBeDefined();
    });

    test("is a function", () => {
        expect(typeof fizzBuzz).toEqual("function");
    });
});

describe("logic", () => {
    xtest("returns 'zero' for input of 0", () => {
        expect(fizzBuzz(0)).toBe("zero");
    });

    xtest("returns fizz for numbers with multiply of 3", () => {
        expect(fizzBuzz(3)).toBe("fizz");
        expect(fizzBuzz(18)).toBe("fizz");
        expect(fizzBuzz(24)).toBe("fizz");
        expect(fizzBuzz(48)).toBe("fizz");
        expect(fizzBuzz(93)).toBe("fizz");
        expect(fizzBuzz(1236)).toBe("fizz");
    });

    xtest("returns buzz for numbers with multiply of 5", () => {
        expect(fizzBuzz(5)).toBe("buzz");
        expect(fizzBuzz(20)).toBe("buzz");
        expect(fizzBuzz(50)).toBe("buzz");
        expect(fizzBuzz(80)).toBe("buzz");
        expect(fizzBuzz(200)).toBe("buzz");
        expect(fizzBuzz(1000)).toBe("buzz");
    });

    xtest("returns fizzbuzz for numbers with multiply of 3 and 5", () => {
        expect(fizzBuzz(15)).toBe("fizzbuzz");
        expect(fizzBuzz(30)).toBe("fizzbuzz");
        expect(fizzBuzz(45)).toBe("fizzbuzz");
        expect(fizzBuzz(600)).toBe("fizzbuzz");
        expect(fizzBuzz(1500)).toBe("fizzbuzz");
        expect(fizzBuzz(6000)).toBe("fizzbuzz");
    });

    xtest("returns input number for other inputs", () => {
        expect(fizzBuzz(4)).toBe(4);
        expect(fizzBuzz(1)).toBe(1);
        expect(fizzBuzz(17)).toBe(17);
        expect(fizzBuzz(442)).toBe(442);
        expect(fizzBuzz(13)).toBe(13);
        expect(fizzBuzz(148)).toBe(148);
        expect(fizzBuzz(202)).toBe(202);
    });
});

File: interview-main/src/python/fizz_buzz/fizz_buzz_test.py
import unittest

from fizz_buzz import (
    fizzbuzz,
)

class FizzBuzzTest(unittest.TestCase):
    def test_returns_zero_for_input_of_zero(self):
        self.assertEqual(fizzbuzz(0), "zero")

    def test_returns_fizz_for_numbers_with_multiply_of_3(self):
        self.assertEqual(fizzbuzz(3), "fizz")
        self.assertEqual(fizzbuzz(18), "fizz")
        self.assertEqual(fizzbuzz(24), "fizz")
        self.assertEqual(fizzbuzz(48), "fizz")
        self.assertEqual(fizzbuzz(93), "fizz")
        self.assertEqual(fizzbuzz(1236), "fizz")

    def test_returns_buzz_for_numbers_with_multiply_of_5(self):
        self.assertEqual(fizzbuzz(5), "buzz")
        self.assertEqual(fizzbuzz(20), "buzz")
        self.assertEqual(fizzbuzz(50), "buzz")
        self.assertEqual(fizzbuzz(80), "buzz")
        self.assertEqual(fizzbuzz(200), "buzz")
        self.assertEqual(fizzbuzz(1000), "buzz")

    def test_returns_fizzbuzz_for_numbers_with_multiply_of_3_and_5(self):
        self.assertEqual(fizzbuzz(15), "fizzbuzz")
        self.assertEqual(fizzbuzz(30), "fizzbuzz")
        self.assertEqual(fizzbuzz(45), "fizzbuzz")
        self.assertEqual(fizzbuzz(600), "fizzbuzz")
        self.assertEqual(fizzbuzz(1500), "fizzbuzz")
        self.assertEqual(fizzbuzz(6000), "fizzbuzz")

    def test_returns_input_number_for_other_input(self):
        self.assertEqual(fizzbuzz(4), 4)
        self.assertEqual(fizzbuzz(1), 1)
        self.assertEqual(fizzbuzz(17), 17)
        self.assertEqual(fizzbuzz(442), 442)
        self.assertEqual(fizzbuzz(13), 13)
        self.assertEqual(fizzbuzz(148), 148)
        self.assertEqual(fizzbuzz(202), 202)
